purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallPresentedCache.kt

```diff
@@ -0,0 +1,38 @@
+package com.revenuecat.purchases.paywalls
+
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.common.verboseLog
+import com.revenuecat.purchases.paywalls.events.PaywallEvent
+import com.revenuecat.purchases.paywalls.events.PaywallEventType
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+internal class PaywallPresentedCache {
+    @get:Synchronized
+    @set:Synchronized
+    private var lastPaywallImpressionEvent: PaywallEvent? = null
+
+    fun getAndRemovePresentedEvent(): PaywallEvent? {
+        val event = lastPaywallImpressionEvent
+        lastPaywallImpressionEvent = null
+        return event
```

Comment:
This is technically a race condition. Is it possible to synchronize `getAndRemovePresentedEvent` instead?

Reply:
Right! We can synchronize the methods instead. Great catch!